### DISCUSSION TITLE
Mermaid's fontawesome support

### DISCUSSION THREAD

**Discussion by kjohnsen** [quarto-dev/quarto-cli#2823] [Q&A] [answered]
It would be awesome to be able to use [Mermaid's fontawesome feature in flowcharts](https://mermaid-js.github.io/mermaid/#/flowchart?id=basic-support-for-fontawesome). It's not working for me--not sure if that's because it was introduced after Quarto forked Mermaid or because of how Mermaid is configured for Quarto.

**Comment by cscheid** [comment, marked as answer]
This is tricky. the font-awesome icons are being generated correctly by Quarto, but somewhere the font awesome fonts need to be actually set on your HTML.

We do have fontawesome support in quarto documents through an [extension](https://github.com/quarto-ext/fontawesome). You need to use the shortcode in the document to enable the extension, so you could try:

````
```{mermaid}
flowchart TD
    B["fab:fa-twitter for peace"]
    B-->C[fa:fa-ban forbidden]
    B-->D(fa:fa-spinner);
    B-->E(A fa:fa-camera-retro perhaps?)
```

::: hidden
  {{< fa thumbs-up >}} 
:::
````

Unfortunately, this produces the following:

<img width="443" alt="image" src="https://user-images.githubusercontent.com/285675/195445251-3c4d5fab-841b-4c89-af36-674b4f3f4bad.png">

As you can see the rectangle sizing calculations from mermaid are not correct. I believe this is happening because we're loading the fontawesome fonts and CSS _after_ mermaid computes its bounding boxes. This kind of multi-library coordination is tricky to solve in general, so I'm not sure how to go about it.
